### ISSUE TITLE
set mykey "my binary safe value"    ... will not work

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ashish21590/redis#286]
```
What steps will reproduce the problem?
1. I open the CLI for redis 1.2.6
2. type the following:
   set mykey "my binary safe value"
3. I get the following error:
   Wrong number of arguments for 'set'

What is the expected output? What do you see instead?
   I expected to get  OK

What version of the product are you using? On what operating system?
Mac OS X 10.5.8, Redis 1.2.6

Please provide any additional information below.

I am a redis newbie, perhaps I am missing something totally obvious .. ?

```

Original issue reported on code.google.com by `galar...@gmail.com` on 17 Jul 2010 at 3:10

**Comment by GoogleCodeExporter** [comment]
```
Did you type the double quotes? With double quotes it is working, without it 
doesn't work because the redis-cli seems not to correctly detect commands: it 
doesn't send the space delimited words as data but as command parameters
```

Original comment by `tobias_p...@hotmail.com` on 17 Jul 2010 at 9:23

**Comment by GoogleCodeExporter** [comment]
```
I have the same issue. On OSX 10.6.4, Redis 1.2.6 installed via 'brew' which 
pulled the archive: http://redis.googlecode.com/files/redis-1.2.6.tar.gz

I'm using double quotes but still get the error. Perhaps a bug catching the 
quotes on OSX?

My exact output:

$ redis-cli 
redis> set mykey "my binary safe value"
Wrong number of arguments for 'set'
redis> 
```

Original comment by `rat...@gmail.com` on 28 Jul 2010 at 3:06

**Comment by GoogleCodeExporter** [comment]
```
This is now fixed in both Redis 2.0.0 and Redis master (will be 2.2.x).

Cheers,
Salvatore
```

Original comment by `anti...@gmail.com` on 30 Aug 2010 at 2:03
- Changed state: **Verified**
